lib/Common/PlatformAgnostic/Arrays.h

```diff
@@ -0,0 +1,35 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+namespace PlatformAgnostic
+{
+namespace Arrays
+{
+    class Utility
+    {
+#ifndef ENABLE_GLOBALIZATION
+        class ArraysLocale
+        {
+            WCHAR commaSeparator;
+            WCHAR semicolonSeparator;
+            WCHAR localeSeparator;
+
+        public:
+
+            ArraysLocale();
+
+            inline WCHAR GetLocaleSeparator() { return localeSeparator; }
+        };
+
+        static ArraysLocale arraysLocale;
+#endif
+    public:
+
+        static size_t GetLocaleSeparator(WCHAR* szSeparator);
```

Comment:
I think I would still prefer that this method took a size parameter to make it clear at call sites what is required. I also would toss out the idea of having the written length be an out pointer so the function can return a bool (with `false` on failure), but I feel less strongly about that.

Reply:
Still figuring out how to resolve this. We could return the length, and pass the buffer size instead as a parameter, to make it somewhat clearer. 
`size_t sepSize = Arrays::GetLocaleSeparator(szSeparator, sepBufSize);`
We could also return an object (`Arrays::Separator`) holding the value and the size of the separator.
Or, as you mentioned, return boolean and use an out pointer.